x-pack/plugins/ingest_manager/server/services/saved_object.ts

```diff
@@ -12,3 +14,63 @@
 export function escapeSearchQueryPhrase(val: string): string {
   return `"${val.replace(/["]/g, '"')}"`;
 }
+
+// Adds `.attribute` to any kuery strings that are missing it, this comes from
+// internal SO structure. Kuery strings that come from UI will typicall have
+// `.attribute` hidden to simplify UX, so this normalizes any kuery string for
+// filtering SOs
+export const normalizeKuery = (savedObjectType: string, kuery: string): string => {
+  return kuery.replace(
+    new RegExp(`${savedObjectType}\.(?!attributes\.)`, 'g'),
+    `${savedObjectType}.attributes.`
+  );
+};
+
+// Like saved object client `.find()`, but ignores `page` and `perPage` parameters and
+// returns *all* matching saved objects by collocating results from all `.find` pages
+export const findAllSOs = async <T = unknown>(
+  soClient: SavedObjectsClientContract,
+  options: Omit<ListWithKuery, 'page' | 'perPage'> & {
+    type: string;
+  }
+): Promise<Pick<SavedObjectsFindResponse<T>, 'saved_objects' | 'total'>> => {
+  const { type, sortField, sortOrder, kuery } = options;
+  let savedObjectResults: SavedObjectsFindResponse<T>['saved_objects'] = [];
+
+  // TODO: This is the default `index.max_result_window` ES setting, which dictates
+  // the maximum amount of results allowed to be returned from a search. It's possible
+  // for the actual setting to differ from the default. Can we retrieve the real
+  // setting in the future?
+  const searchLimit = 10000;
+
+  const query = {
+    type,
+    sortField,
+    sortOrder,
+    filter: kuery,
+    page: 1,
+    perPage: searchLimit,
+  };
+
+  const { saved_objects: initialSOs, total } = await soClient.find<T>(query);
+
+  savedObjectResults = initialSOs;
+
+  if (total > searchLimit) {
+    const remainingPages = Math.ceil((total - searchLimit) / searchLimit);
+    for (let currentPage = 2; currentPage <= remainingPages + 1; currentPage++) {
+      const { saved_objects: currentPageSavedObjects } = await soClient.find<T>({
+        ...query,
+        page: currentPage,
+      });
+      if (currentPageSavedObjects.length) {
+        savedObjectResults = [...savedObjectResults, ...currentPageSavedObjects];
```

Comment:
Can we use an area method like .concat or .push here? Spreading thousands of objects is taxing on the engine